sway-core/src/transform/to_parsed_lang/convert_parse_tree.rs

```diff
@@ -2699,15 +2722,19 @@ fn path_root_opt_to_bool(
     _context: &mut Context,
     handler: &Handler,
     root_opt: Option<(Option<AngleBrackets<QualifiedPathRoot>>, DoubleColonToken)>,
+    throw_error_on_qualified_path_root: bool,
```

Comment:
I don't like this kind of specialization because it's hard to see what this does at a glance.
Consider using a wrapper function to let the caller explicitly pick if they want the error or not. Or throwing the error all the time but handling it with `unwrap_or` at the call site.

Reply:
Removed the parameter `throw_error_on_qualified_path_root` as suggested and the place that was depending on it now uses `path_root_opt_to_bool_and_qualified_path_root`